### PR TITLE
algorithm: percentage of letter

### PR DESCRIPTION
**Pull request by Saimon398**
[Open in Gitpod](https://gitpod.io/from-referrer/)&nbsp;[know more](https://www.gitpod.io/docs/pull-requests/)

### Describe your change:

- [x] Add an algorithm?
- [ ] Fix a bug or typo in an existing algorithm?
- [ ] Documentation change?

### Checklist:

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Javascript/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized.
- [x] I know that pull requests will not be merged if they fail the automated tests.
- [x] This PR only changes one algorithm file. To ease review, please open separate PRs for separate algorithms.
- [x] All new JavaScript files are placed inside an existing directory.
- [x] All filenames should use the UpperCamelCase (PascalCase) style. There should be no spaces in filenames.
      **Example:**`UserProfile.js` is allowed but `userprofile.js`,`Userprofile.js`,`user-Profile.js`,`userProfile.js` are not
- [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
- [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/JavaScript/pull/1261"></a>

### CODE REVIEW

File: String/PercentageOfLetters.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ */
+const percentageOfLetter = (str, letter) => {
+  if (typeof str !== 'string' || typeof letter !== 'string') {
+    throw new Error('Input data must be strings')
+  }
+  let letterCount = 0
+  // Iterate through the whole given text
+  for (let i = 0; i < str.length; i++) {
+    // Count how often the letter appears in the word
+    letterCount = str[i].toLowerCase() === letter.toLowerCase() ? letterCount + 1 : letterCount + 0
```
Comment (appgurueu) [inline String/PercentageOfLetters.js:21]: You can use `if` or `+=` here.

File: String/PercentageOfLetters.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  }
+  const percentage = Math.floor((100 * letterCount) / str.length)
+
```
Comment (appgurueu) [inline String/PercentageOfLetters.js:24]: Why the empty line?

Comment (Saimon398) [reply String/PercentageOfLetters.js]: Fixed

File: String/PercentageOfLetters.js (hunk 1 above)
Comment (Saimon398) [reply String/PercentageOfLetters.js]: All done

**Review by appgurueu — APPROVED** [review]

**Review by raklaptudirm — APPROVED** [review]

**Pull request merged**